src/main/java/org/opensearch/flowframework/workflow/ModelGroupStep.java

```diff
@@ -0,0 +1,113 @@
+import static org.opensearch.flowframework.common.CommonValue.NAME_FIELD;
+
+/**
+ * Step to register a model group
+ */
+public class ModelGroupStep implements WorkflowStep {
+
+    private static final Logger logger = LogManager.getLogger(RegisterModelStep.class);
+
+    private MachineLearningNodeClient mlClient;
+
+    static final String NAME = "model_group";
+
+    /**
+     * Instantiate this class
+     * @param mlClient client to instantiate MLClient
+     */
+    public ModelGroupStep(MachineLearningNodeClient mlClient) {
+        this.mlClient = mlClient;
+    }
+
+    @Override
+    public CompletableFuture<WorkflowData> execute(List<WorkflowData> data) throws IOException {
+
+        CompletableFuture<WorkflowData> registerModelGroupFuture = new CompletableFuture<>();
+
+        ActionListener<MLRegisterModelGroupResponse> actionListener = new ActionListener<>() {
+            @Override
+            public void onResponse(MLRegisterModelGroupResponse mlRegisterModelGroupResponse) {
+                logger.info("Model group registration successful");
+                registerModelGroupFuture.complete(
+                    new WorkflowData(
+                        Map.ofEntries(
+                            Map.entry("model_group_id", mlRegisterModelGroupResponse.getModelGroupId()),
+                            Map.entry("model_group_status", mlRegisterModelGroupResponse.getStatus())
+                        )
+                    )
+                );
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                logger.error("Failed to register model group");
+                registerModelGroupFuture.completeExceptionally(new FlowFrameworkException(e.getMessage(), ExceptionsHelper.status(e)));
+            }
+        };
+
+        String modelGroupName = null;
+        String description = null;
+
+        for (WorkflowData workflowData : data) {
+            Map<String, Object> content = workflowData.getContent();
+
+            for (Entry<String, Object> entry : content.entrySet()) {
+                switch (entry.getKey()) {
+                    case NAME_FIELD:
+                        modelGroupName = (String) content.get(NAME_FIELD);
+                        break;
+                    case DESCRIPTION:
+                        description = (String) content.get(DESCRIPTION);
+                        break;
+                    default:
+                        break;
+                }
+            }
+        }
+
+        if (Stream.of(modelGroupName, description).allMatch(x -> x != null)) {
+            MLRegisterModelGroupInput mlInput = MLRegisterModelGroupInput.builder().name(modelGroupName).description(description).build();
+
+            mlClient.registerModelGroup(mlInput, actionListener);
+        } else {
+            registerModelGroupFuture.completeExceptionally(
+                new FlowFrameworkException("Required fields are not provided", RestStatus.BAD_REQUEST)
```

Comment:
nit: can be done later but should we move common error messages like these to a common file so we have consistent grammar, punctuation and easier for testing if we change anything.

Reply:
Made the exception error specific to this step.